File: CSharpPrototype/PausableTimer.cs
namespace Work_Timer_Box
{
    public class PausableTimer : Timer
    {
        public long StoredMinutes { get; private set; }
        public long StoredSeconds { get; private set; }

        public void PauseTimer()
        {
            if (IsRunning)
            {
                IsRunning = false;
                StoredMinutes += UnixTimeToMinutes(base.GetElapsed());
                StoredSeconds += UnixTimeToSeconds(base.GetElapsed());
            }
        }

        public override long GetElapsed()
        {
            return base.GetElapsed() + 60 * StoredMinutes + StoredSeconds;
        }

        public override long GetElapsedMinutes()
        {
            return UnixTimeToMinutes(base.GetElapsed() + StoredSeconds) + StoredMinutes;
        }

        public override long GetElapsedSeconds()
        {
            return UnixTimeToSeconds(base.GetElapsed() + StoredSeconds);
        }

        public override void ResetTimer()
        {
            base.ResetTimer();
            StoredMinutes = 0;
            StoredSeconds = 0;
        }
    }
}

File: Pico/pausable_timer.cpp
#include "consulting_clock.hpp"

void PausableTimer::PauseTimer()
{
    if (isRunning)
    {
        isRunning = false;
        
        storedMinutes += Timer::GetElapsedMinutes();
        storedSeconds += Timer::GetElapsedSeconds();

        storedMinutes += storedSeconds / SEC_TO_MIN;
        storedSeconds %= SEC_TO_MIN;
    }
}

uint64_t PausableTimer::GetElapsed()
{
    return Timer::GetElapsed() + storedMinutes * US_TO_MIN + storedSeconds * US_TO_SEC;
}

uint16_t PausableTimer::GetElapsedMinutes()
{
    return PausableTimer::GetElapsed() / US_TO_MIN;
}

uint16_t PausableTimer::GetElapsedSeconds()
{
    return PausableTimer::GetElapsed() % US_TO_MIN / US_TO_SEC;
}

void PausableTimer::ResetTimer()
{
    Timer::ResetTimer();
    storedMinutes = 0;
    storedSeconds = 0;
}
